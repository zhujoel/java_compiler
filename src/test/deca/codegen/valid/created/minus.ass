; Test pour savoir si la pile est pleine
	TSTO #27
	BOV pile_pleine
	ADDSP #25
; start main program
; Class Declaration
; Code de la table des méthodes de Object
	LOAD #null, R0
	STORE R0, 19(GB)
	LOAD code.Object.equals, R0
	STORE R0, 20(GB)
; Main Function
; Declaration des variables
	LOAD #5, R2
	STORE R2, 21(GB)
	LOAD #2, R2
	STORE R2, 22(GB)
	LOAD #0x0.0p0, R0
	STORE R0, 23(GB)
	LOAD #0x0.0p0, R0
	STORE R0, 24(GB)
; Beginning of main instructions:
; -
	LOAD 21(GB), R2
	LOAD 22(GB), R3
	SUB R2, R3
	BOV debordement
; -
	LOAD #6, R2
	PUSH R3
	LOAD #1, R3
	SUB R2, R3
	BOV debordement
; -
	LOAD 21(GB), R2
	PUSH R3
	LOAD 22(GB), R3
	SUB R3, R2
	BOV debordement
	POP R3
	LOAD R2, R1
	WINT
; -
	LOAD 21(GB), R3
	PUSH R3
	LOAD 22(GB), R3
	SUB R3, R3
	BOV debordement
	POP R3
	STORE R3, 21(GB)
	POP R3
; -
	LOAD #5, R3
	PUSH R3
	LOAD #2, R3
	SUB R3, R3
	BOV debordement
	POP R3
	LOAD R3, R1
	WINT
	LOAD #5, R3
	FLOAT R3, R3
	STORE R3, 23(GB)
	LOAD #2, R3
	FLOAT R3, R3
	STORE R3, 24(GB)
; -
	LOAD 23(GB), R3
	PUSH R3
	LOAD 24(GB), R3
	SUB R3, R3
	BOV debordement
	POP R3
; -
	LOAD #0x1.8p2, R3
	PUSH R3
	LOAD #0x1.0p0, R3
	SUB R3, R3
	BOV debordement
	POP R3
; -
	LOAD 23(GB), R3
	PUSH R3
	LOAD 24(GB), R3
	SUB R3, R3
	BOV debordement
	POP R3
	LOAD R3, R1
	WINT
; -
	LOAD 23(GB), R3
	PUSH R3
	LOAD 24(GB), R3
	SUB R3, R3
	BOV debordement
	POP R3
	STORE R3, 23(GB)
; -
	LOAD #0x1.4p2, R3
	PUSH R3
	LOAD #0x1.0p1, R3
	SUB R3, R3
	BOV debordement
	POP R3
	LOAD R3, R1
	WINT
	HALT
code.Object.equals:
pile_pleine:
	WSTR "Erreur : Pile pleine."
	WNL
	ERROR
dereferencement_null:
	WSTR "Erreur : dereferencement null."
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : Tas plein."
	WNL
	ERROR
debordement:
	WSTR "Erreur : debordement durant une operation arithmétique"
	WNL
	ERROR
io_error:
	WSTR "Erreur : input ou output"
	WNL
	ERROR
; end main program
