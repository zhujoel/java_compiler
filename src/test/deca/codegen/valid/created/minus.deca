{
    int a = 5;
    int b = 2;
    float c;
    float d;
    a - b;
    6 - 1;
    println(a - b);
    a = a - b;
    println(5 - 2);
    c = 5;
    d = 2;
    c - d;
    6.0 - 1.0;
    println(c - d);
    c = c - d;
    println(5.0 - 2.0);
}
