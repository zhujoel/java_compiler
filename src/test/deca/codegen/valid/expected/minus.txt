3
3
3.00000e+00
3.00000e+00
